assets/js/googlesitekit/data/create-settings-store.test.js

```diff
@@ -393,6 +393,135 @@ describe( 'createSettingsStore store', () => {
 				dispatch.setSettings( serverValues );
 				expect( select.haveSettingsChanged() ).toEqual( false );
 			} );
+
+			it( 'compares all keys when keys argument is not supplied', async () => {
+				const serverValues = {
+					setting1: 'serverside',
+					setting2: 'test-value',
+				};
+				const clientValues = {
+					setting1: 'clientside',
+					setting2: 'test-value',
+				};
+
+				fetchMock.getOnce(
+					/^\/google-site-kit\/v1\/core\/site\/data\/settings/,
```

Comment:
Non-essential but I think worth a mention - when testing for paths, I like to construct the regex via a call to `RegExp`, that way the path separator doesn't need to be escaped, and the path is more easily read - and searched for.

I know we have got a majority of path regexes using the inline style, but we do use `RegExp` in a number of places and I think it's worth promoting for readability.

That said - I do realise there are other path regexes within this file so you might prefer to keep your additions as-is for consistency within the file. Leaving this comment for your digestion.

```suggestion
					RegExp( '^/google-site-kit/v1/core/site/data/settings' ),
```

Reply:
Hmm, given right now we only use the `new RegExp` constructor when we have variables inside the regex, I'm inclined to leave this as-is here.

I find both slightly awkward and am _personally_ used to writing regexes with the escapes, but either works for me. 😅 

Happy to make the change across the repo but can you file an issue instead? If there's an ESLint rule or something to tackle it/enforce it/mass-fix the existing regexes I'd be cool with this, but I don't want to do it piece-meal for now as currently it's at least consistent 🙂

Reply:
No probs - I will create an issue for it, sounds good :+1: